scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowFunctions.scala

```diff
@@ -283,43 +285,56 @@ class TFExampleSCollectionFunctions[T <: Example](val self: SCollection[T]) {
   /**
    * Save this SCollection of `org.tensorflow.example.Example` as a TensorFlow TFRecord file,
    * along with a protobuf file representing the inferred `org.tensorflow.metadata.v0.Schema`.
+   * If you need to alter the schema, use [[inferExampleMetadata]] and pass altered schema to
+   * [[saveAsTfExampleFile()]].
    * @group output
    */
-  def saveAsTfExampleFileWithMetadata(path: String,
-                                      schemaSuffix: String = "schema.pb",
-                                      suffix: String = ".tfrecords",
-                                      compression: Compression = Compression.UNCOMPRESSED,
-                                      numShards: Int = 0)
-                                     (implicit ev: T <:< Example): Future[Tap[Example]] = {
-    import scala.concurrent.ExecutionContext.Implicits.global
+  def saveAsTfExampleFile(path: String)(implicit ev: T <:< Example)
+  : (Future[Tap[Example]], Future[Tap[Schema]]) = {
+    this.saveAsTfExampleFile(
+      path,
+      schema = null,
+      schemaFilename = "_schema.pb",
+      suffix = ".tfrecords",
+      compression = Compression.UNCOMPRESSED,
+      numShards = 0
+    )(ev)
+  }
 
-    val schemaPath = path.replaceAll("\\/+$", "") + schemaSuffix
-    buildAndSaveExampleMetadata(schemaPath)(ev)
-    val r = self.map(_.toByteArray).saveAsTfRecordFile(path, suffix, compression, numShards)
-    r.map(_.map(Example.parseFrom))
+  def saveAsTfExampleFile(path: String,
+                          schema: SCollection[Schema],
+                          schemaFilename: String,
+                          suffix: String,
+                          compression: Compression,
+                          numShards: Int)
+                          (implicit ev: T <:< Example)
+  : (Future[Tap[Example]], Future[Tap[Schema]]) = {
+    val inferedSchema = Option(schema).getOrElse(self.inferExampleMetadata(ev))
+    val schemaPath = path.replaceAll("\\/+$", "") + "/" + schemaFilename
```

Comment:
@ravwojdyla can we express this in a better way? can we use string interpolation?